app/components/inputs/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,143 @@
+import { ImageListType, ImageType } from "react-images-uploading";
+import { useState } from "react";
+import axios from "axios";
+import { toast } from "react-hot-toast";
+import { PreviewImageWithUrl } from "@/app/types";
+
+interface Props {
+  value: PreviewImageWithUrl[];
+  onChange: (value: PreviewImageWithUrl[]) => void;
+}
+
+export const useUploadImage = ({ value: images, onChange }: Props) => {
+  const [previewImages, setPreviewImages] = useState(
+    images.map((image) => {
+      const { url, ...rest } = image;
+      return { ...rest, dataURL: url as string } as ImageType;
+    })
+  );
+
+  const [isUploading, setIsUploading] = useState(false);
+
+  const uploadImage = async (imageData: string) => {
+    const axiosResponse = await axios.post("/api/images", {
+      imageData,
+    });
+    return axiosResponse.data.url as string;
+  };
+
+  const onImageFileChange = async (
+    imageList: ImageListType,
+    addUpdateIndex: number[] | undefined
+  ) => {
+    console.log(imageList);
+    setPreviewImages(imageList);
+
+    // delete image
+    if (addUpdateIndex === undefined) {
+      const remainingImages = images.filter((image) =>
+        imageList.find((v) => v.dataURL === image.url)
+      );
+      onChange(remainingImages);
+      setPreviewImages(
+        remainingImages.map((image) => {
+          const { url, ...rest } = image;
+          return { ...rest, dataURL: url as string };
+        })
+      );
+      return;
```

Comment:
minor thing. no need to write `return` at the end of a function, as it's `return` by default

Reply:
I don't know if you've heard of guard clauses, that's what I'm doing here. It basically works the same as if-else blocks, once I'm done with a case, I return so I break out of the function. Many js/ts programmers prefer this over if-else blocks for reasons including 1) you're sure that you've got all the logic for a case being returned early and you can completely forget about it and deal with other cases; 2) the last case doesn't need to be nested; 3) there can be a line between each two cases so it looks cleaner especially with comments

Reply:
my bad, I misread the scope